packages/frontend/src/context/UserContext.tsx

```diff
@@ -0,0 +1,304 @@
+// SPDX-FileCopyrightText: 2024 Mass Labs
+//
+// SPDX-License-Identifier: GPL-3.0-or-later
+
+import React, { createContext, useContext, useEffect, useState } from "react";
+import { useEnsAvatar, useWalletClient } from "wagmi";
+import {
+  RelayClient,
+  discoverRelay,
+  type WalletClientWithAccount,
+} from "@massmarket/client";
+import { hardhat, mainnet, sepolia } from "viem/chains";
+import { http, createPublicClient, createWalletClient } from "viem";
+import { useAuth } from "@/context/AuthContext";
+import * as abi from "@massmarket/contracts";
+import { Status, ShopId, ObjectType } from "@/types";
+import { type ClientContext } from "@/context/types";
+import { privateKeyToAccount } from "viem/accounts";
+import { usePathname } from "next/navigation";
+import { random32BytesHex } from "@massmarket/utils";
+import { useSearchParams } from "next/navigation";
+import debugLib from "debug";
+
+export const UserContext = createContext<ClientContext>({
+  walletAddress: null,
+  ensName: null,
+  shopId: "0x",
+  avatar: null,
+  relayClient: null,
+  shopPublicClient: null,
+  inviteSecret: null,
+  clientWallet: null,
+  createNewRelayClient: () =>
+    new Promise(() => {
+      return null;
+    }),
+  setShopId: () => {},
+  setInviteSecret: () => {},
+  setRelayClient: () => {},
+  setWallet: () => {},
+  checkPermissions: () =>
+    new Promise(() => {
+      return false;
+    }),
+  sendGuestCheckoutSubscription: () => new Promise(() => {}),
```

Comment:
```suggestion
  upgradeGuestToCustomer: () => new Promise(() => {}),
```

since it also enrolls the keycard / it's not just the subscription.